server/src/main.rs

```diff
@@ -367,4 +395,24 @@ mod tests {
             "[\"file-a\",\"file-b\",\"file-c\",\"nested-dir\"]"
         );
     }
+
+    // resolve tests
+
+    #[test]
+    fn resolve_ref_with_empty_repo() {
+        assert_test_server_responds_with!(
+            "/repos/fixtures/resolve?reference=467e981f94686d7a1db395f8acfd3cf7e7adfcd3",
+            200,
+            "467e981f94686d7a1db395f8acfd3cf7e7adfcd3"
+        )
+    }
+
+    #[test]
+    fn resolve_ref_with_invalid_ref() {
```

Comment:
Missng some happy path tests here when you give a branch or tag I think, to make sure you get a valid SHA back.